[Tan ZiYang Henry] iP

Tanziyanghenry opened the pull request:
### Henry

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Henry frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~EASY~~ SUPER EASY to use

All you need to do is,

1. download it from [here](https://github.com/Tanziyanghenry/ip/releases/tag/A-Jar).
2. copy the jar file into an empty folder.
3. open a command window in that folder.
4. run the command `java -jar "henry.jar"` (i.e., run the command in the same folder as the jar file).
5. tell Henry what task you want to record.

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Tanziyanghenry committed:
Rename main class

Tanziyanghenry committed:
Add greeting

Tanziyanghenry committed:
Add support for echo

Tanziyanghenry committed:
Add support to list

Tanziyanghenry committed:
add Task class

Tanziyanghenry committed:
Replace array of String to array of Task

Tanziyanghenry committed:
Add methods printList and addTask

Tanziyanghenry committed:
Add mark tasks as done

Tanziyanghenry committed:
Add Todo, Event and Deadline class

Tanziyanghenry committed:
Add toString method in Task class

Tanziyanghenry committed:
Modify addTask method in main class

Tanziyanghenry committed:
Add HenryException

Tanziyanghenry committed:
Handle exception where input is not a task

Tanziyanghenry committed:
Handle exception when the task description is incomplete

Tanziyanghenry committed:
Add more specific explanation of error

Tanziyanghenry committed:
Handle exception when user input is list but there is no task

Tanziyanghenry committed:
Add isDone method in Task class

Tanziyanghenry committed:
Handle exception when task which is marked is asked to be mark again, likewise for unmark

Tanziyanghenry committed:
Account for case sensitive cases

Tanziyanghenry committed:
Handle exception for if number is within range

Tanziyanghenry committed:
Handle exception when a non integer is given to mark or unmark task

Tanziyanghenry committed:
Add comments to describe the exception

Tanziyanghenry committed:
Replace Task[] with ArrayList<Task>

Tanziyanghenry committed:
Add support for deleting task

Tanziyanghenry committed:
Edit description for invalid number exception

Tanziyanghenry committed:
TextUiTesting

Tanziyanghenry committed:
Add comments in JavaDoc format

Tanziyanghenry committed:
Add summary method in Todo, Deadline, Event and Task class

Tanziyanghenry committed:
Check if file exist in computer

Tanziyanghenry committed:
Add writeToFile method in main

Tanziyanghenry committed:
Overload constructors

Tanziyanghenry committed:
Load data from hard disk

Tanziyanghenry committed:
Merge branch 'branch-Level-7'

Tanziyanghenry committed:
Add conversion of date and time

Tanziyanghenry committed:
Merge branch 'branch-Level-8'

Tanziyanghenry committed:
Fix comment in main class

Tanziyanghenry committed:
Fix comment in Deadline class

Tanziyanghenry committed:
Fix comments in Event and Todo class

Tanziyanghenry committed:
Add Storage, TaskList, Ui classes

Tanziyanghenry committed:
Add Command classes

Tanziyanghenry committed:
Add Parser Class

Tanziyanghenry committed:
Implement new classes into main class

Tanziyanghenry committed:
Merge branch 'branch-MoreOOP'

Tanziyanghenry committed:
Add henry package

Tanziyanghenry committed:
Add henry.command package

Tanziyanghenry committed:
Add henry.task package

Tanziyanghenry committed:
Merge branch 'branch-Packages'

Tanziyanghenry committed:
Merge branch 'add-gradle-support'

Tanziyanghenry committed:
Fix bug in Ui class

Tanziyanghenry committed:
Fix bugs in classes

Tanziyanghenry committed:
Update text-ui-test

Tanziyanghenry committed:
Add util package

Tanziyanghenry committed:
Remove save method in execute method

Tanziyanghenry committed:
Build gradle

Tanziyanghenry committed:
Merge branch 'branch-Gradle'

Tanziyanghenry committed:
Add JUnit tests

Tanziyanghenry committed:
Merge branch 'Branch-A-JUnit'

Tanziyanghenry committed:
Edit build.gradle

Tanziyanghenry committed:
Adhere to coding standard

Tanziyanghenry committed:
Merge branch 'branch-A-CodingStandard'

Tanziyanghenry committed:
Add javadoc comments

Tanziyanghenry committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/henry/command/AddCommand.java

Tanziyanghenry committed:
Edit imports in AddCommand

Tanziyanghenry committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/henry/command/AddCommand.java

Tanziyanghenry committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/henry/command/AddCommand.java

Tanziyanghenry committed:
Edit input to test more cases

Tanziyanghenry committed:
Add find command in Parser class

Tanziyanghenry committed:
Add FindCommand class

Tanziyanghenry committed:
Merge branch 'branch-Level-9'

Tanziyanghenry committed:
Improve coding standard in FindCommand

Tanziyanghenry committed:
Improve coding style

Tanziyanghenry committed:
Merge branch 'branch-A-CheckStyle'

Tanziyanghenry committed:
add javafx

Tanziyanghenry committed:
Add GUI to chatbot

Tanziyanghenry committed:
Merge branch 'branch-Level-10'

Clarenceeey commented on src/main/java/Command.java:
```diff
@@ -0,0 +1,50 @@
+
+    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws HenryException;
+
+    public boolean isExit(){
```
Shouldn't there be a space between before the opening bracket?
```suggestion
    public boolean isExit() {
```

Clarenceeey commented on src/main/java/Storage.java:
```diff
@@ -0,0 +1,77 @@
+     */
+    public ArrayList<Task> load() throws HenryException {
+        // create a Scanner using the File as the source
+        try{
```
Shouldn't there be a space before the opening bracket?
```suggestion
        try {
```

Clarenceeey commented on src/main/java/henry/command/Command.java:
```diff
@@ -0,0 +1,58 @@
+
+    public abstract String execute(TaskList tasks, Ui ui) throws HenryException;
+
+    public boolean isExit() {
```
Perhaps a javadocs here would be nice?

Clarenceeey commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+
+    protected String by;
+
+    public Deadline(String description, String by) {
```
Javadocs here could be used to explain the different use cases of the overloaded constructor

Clarenceeey reviewed (commented):
Looks good to me 🎉

Al-ez commented on src/main/java/Todo.java:
```diff
@@ -0,0 +1,30 @@
+     * @return task summary
+     */
+    @Override
+    public String summary() {
```
nit: for method names try to use verbs instead (from textbook), maybe something like getSummary() will be better

Al-ez commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,41 @@
+     * @return task summary
+     */
+    @Override
+    public String summary() {
```
nit: for method names try to use verbs instead (from textbook), maybe something like getSummary() will be better

Al-ez commented on src/main/java/ExitCommand.java:
```diff
@@ -0,0 +1,23 @@
+public class ExitCommand extends Command {
+
+    @Override
+    public boolean isExit(){
```
nit: for method names try to use verbs instead (from textbook), maybe something like exit() will be better. Nouns are more used for variables/fields instead :)

Al-ez commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,74 @@
+     *
+     * @return the status of the task
+     */
+    public boolean isDone() {
```
nit: for method names try to use verbs instead, maybe something like getCompletionStatus(), or getStatus() or getIsDone() will be better

Al-ez reviewed (commented):
Looks good to me, just a few nit comments :) 😸

Tanziyanghenry committed:
Add assertions to ChangeStatusCommand for improved error handling

Introduced assertions to ensure that the command is either "mark" or
"unmark" before performing any status change on tasks. This ensures that
only valid commands are executed, preventing potential errors caused by
unexpected or invalid inputs.

Tanziyanghenry committed:
Merge pull request #1 from Tanziyanghenry/branch-A-Assertions

Add assertions to ChangeStatusCommand for improved error handling

Tanziyanghenry committed:
Merge pull request #2 from Tanziyanghenry/master

Merge master to branch-A-CodeQuality

Tanziyanghenry committed:
Restructure execute method by logically splitting responsibilities

The code was reorganized by separating the logic for determining the
number of tasks, identifying the task type, and extracting the activity
details. This restructuring improves code readability by logically
separating different parts of the method to make each section easier to
understand and modify.

Tanziyanghenry committed:
Avoid deep nesting by removing unnecessary else statements.

Deep nesting can make the code harder to read and understand. By
eliminating unnecessary else blocks after return statements, the code
becomes cleaner and more straightforward.

Tanziyanghenry committed:
Maintains single level of abstraction by extracting getTask method

This change maintains a single level of abstraction within the execute
method, making the code more readable and organized.

Tanziyanghenry committed:
Maintains single level of abstraction by extracting getNumber method

This change maintains a single level of abstraction within the execute
method, making the code more readable and organized.

Tanziyanghenry committed:
Merge pull request #3 from Tanziyanghenry/branch-A-CodeQuality

Improve code quality

Tanziyanghenry committed:
Edit javadoc comments

Tanziyanghenry committed:
Add javadoc comments

Tanziyanghenry committed:
Use assertion in Storage class

Tanziyanghenry committed:
add mass delete option

Tanziyanghenry committed:
Merge branch 'branch-C-MassOps'

Tanziyanghenry committed:
Enable Assertions

Tanziyanghenry committed:
Shorten execute method

Tanziyanghenry committed:
Improve SLAP in execute method

Tanziyanghenry committed:
Remove redundant code and improve SLAP in execute method

Tanziyanghenry committed:
Shorten load method and edit javadoc comments

Tanziyanghenry committed:
Add punctuation in javadoc comments

Tanziyanghenry committed:
Edit javadoc comments by adding throw exception

Tanziyanghenry committed:
Remove redundant code

Tanziyanghenry committed:
Improve GUI

Tanziyanghenry committed:
Merge branch 'branch-A-BetterGui'

Tanziyanghenry committed:
Handle command format errors

Tanziyanghenry committed:
Merge branch 'branch-A-MoreErrorHandling'

Tanziyanghenry committed:
Edit chat message

Tanziyanghenry committed:
Edit more chat message

Tanziyanghenry committed:
Fix error of finding keyword

Tanziyanghenry committed:
Fix too long method

Tanziyanghenry committed:
Improve coding style

Tanziyanghenry committed:
Improve SLAP in execute method and fix error handling

Tanziyanghenry committed:
Add name at title bar of GUI

Tanziyanghenry committed:
Add javadoc comments

Tanziyanghenry committed:
Remove redundant code

Tanziyanghenry committed:
Edit product name

Tanziyanghenry committed:
Rename variable

Tanziyanghenry committed:
Add user guide

Tanziyanghenry committed:
Merge branch 'branch-A-UserGuide'

Tanziyanghenry committed:
Edit README.md

Tanziyanghenry committed:
Update README.md

Tanziyanghenry committed:
Add screenshot

Tanziyanghenry committed:
Edit README.md by changing relative path
